ui/store/actions.js

```diff
@@ -2231,6 +2231,19 @@ export function setAdvancedGasFee(val) {
   };
 }
 
+export function setEIP1559V2Enabled(val) {
+  return (dispatch) => {
+    dispatch(showLoadingIndication());
+    log.debug(`background.setEIP1559V2Enabled`);
+    background.setEIP1559V2Enabled(val, (err) => {
+      dispatch(hideLoadingIndication());
+      if (err) {
+        dispatch(displayWarning(err.message));
+      }
+    });
+  };
```

Comment:
Nit: Could you switch to using `promisifiedBackground` instead? `background` is deprecated and should no longer be used for new code.

```suggestion
  return async (dispatch) => {
    dispatch(showLoadingIndication());
    log.debug(`background.setEIP1559V2Enabled`);
    try {
      await promisifiedBackground.setEIP1559V2Enabled(val)
    } catch (error) {
      dispatch(displayWarning(error.message));
    } finally {
      dispatch(hideLoadingIndication());
    }
  };
```

Reply:
Also, it's probably better to bubble the error up in this case. The `displayWarning` pattern is awful and should also be considered deprecated - that results in the error being shown directly to the user in various parts of the wallet with no context or steps they can take to rectify it.

This call is not supposed to fail in general, so the better approach would be to let it bubble up unimpeded so that it makes its way to Sentry.


```suggestion
  return async (dispatch) => {
    dispatch(showLoadingIndication());
    log.debug(`background.setEIP1559V2Enabled`);
    try {
      await promisifiedBackground.setEIP1559V2Enabled(val)
    } finally {
      dispatch(hideLoadingIndication());
    }
  };
```